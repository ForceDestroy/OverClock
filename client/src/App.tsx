import { BrowserRouter, Route, Routes, useLocation } from 'react-router-dom';
import { Box, Grommet } from 'grommet';
import './App.css';

import Home from './pages/Home';
import NotFound from './pages/NotFound';
import Login from './pages/Login';
import EmployeeDashboard from './pages/EmployeeDashboard';
import EmployerDashboard from './pages/EmployerDashboard';
import NavigationTopBar from './components/NavigationTopBar';
import ProfilePage from './pages/ProfilePage';
import Timesheets from './pages/Timesheets';
import Schedule from './pages/Schedule';
import TimeOff from './pages/TimeOff';
import EmployerTimesheets from './pages/EmployerTimesheets';
import Payslip from './pages/Payslip';
import EmployerRequests from './pages/EmployerRequests';
import CompanyPolicy from './pages/CompanyPolicy';

export function App() {
  const { pathname } = useLocation();
  const toggleNav = pathname === '/Login' || pathname === '/';
  const availableRoutes = [
    '/employeedashboard',
    '/employeraccount',
    '/employerdashboard',
    '/employer/timesheets',
    '/employer/requests',
    '/application',
    '/applicationreferral',
    '/applicationmanagement',
    '/companydashboard',
    '/employeedashboard/companypolicy',
    '/profilepage',
    '/employeedashboard/timesheets',
    '/employeedashboard/schedule',
    '/employeedashboard/payslip',
    '/employeedashboard/timeoff',
  ];
  const errorRoute = availableRoutes.includes(pathname.toLowerCase());
  return (
    <div className="App">
      <Grommet full>
        {!toggleNav && errorRoute && <NavigationTopBar />}
        <Box direction="row" height={{ min: '90vh' }} background="light-1">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/Login" element={<Login />} />
            <Route path="/EmployeeDashboard" element={<EmployeeDashboard />} />
            <Route path="/EmployerDashboard" element={<EmployerDashboard />} />
            <Route path="/Employer/Requests" element={<EmployerRequests />} />
            <Route
              path="/Employer/Timesheets"
              element={<EmployerTimesheets />}
            />
            <Route
              path="/EmployeeDashboard/CompanyPolicy"
              element={<CompanyPolicy />}
            />
            <Route path="/ProfilePage" element={<ProfilePage />} />
            <Route
              path="/EmployeeDashboard/Timesheets"
              element={<Timesheets />}
            />
            <Route path="/EmployeeDashboard/Schedule" element={<Schedule />} />
            <Route path="/EmployeeDashboard/Payslip" element={<Payslip />} />
            <Route path="/EmployeeDashboard/TimeOff" element={<TimeOff />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </Box>
      </Grommet>
    </div>
  );
}

export function WrappedApp() {
  return (
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
}
