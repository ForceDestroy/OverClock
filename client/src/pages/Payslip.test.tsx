import { render, screen, waitFor, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import axios from 'axios';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, vi } from 'vitest';
import { App } from '../App';

import AuthTokenHelper from '../helpers/AuthTokenHelper';

vi.mock('axios');
vi.mock('../helpers/AuthTokenHelper');

describe('Payslip', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    cleanup();
  });
  afterEach(() => {
    vi.clearAllMocks();
    vi.resetAllMocks();
  });
  it('Renders Payslip to match snapshot', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: [
        {
          address: 'somewhere',
          amountAccumulated: 2,
          ei: 1,
          eiAccumulated: 2,
          endDate: '2023-01-02T00:00:00',
          fit: 1,
          fitAccumulated: 2,
          grossAmount: 1,
          hoursAccumulated: 2,
          hoursWorked: 1,
          issueDate: '2023-01-05T00:00:00',
          name: 'test',
          netAmount: 1,
          netAccumulated: 3,
          qcpit: 1,
          qcpitAccumulated: 2,
          qpip: 1,
          qpipAccumulated: 2,
          qpp: 1,
          qppAccumulated: 2,
          startDate: '2023-01-01T00:00:00',
        },
      ],
      status: 200,
      statusText: 'OK',
    });

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const snap = render(
      <MemoryRouter initialEntries={['/employeedashboard/payslip']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders Payslip for Employee', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: [
        {
          address: 'somewhere',
          amountAccumulated: 2,
          ei: 1,
          eiAccumulated: 2,
          endDate: '2023-01-02T00:00:00',
          fit: 1,
          fitAccumulated: 2,
          grossAmount: 1,
          hoursAccumulated: 2,
          hoursWorked: 1,
          issueDate: '2023-01-05T00:00:00',
          name: 'test',
          netAmount: 1,
          netAccumulated: 3,
          qcpit: 1,
          qcpitAccumulated: 2,
          qpip: 1,
          qpipAccumulated: 2,
          qpp: 1,
          qppAccumulated: 2,
          startDate: '2023-01-01T00:00:00',
        },
      ],
      status: 200,
      statusText: 'OK',
    });

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    render(
      <MemoryRouter initialEntries={['/employeedashboard/payslip']}>
        <App />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Payslips')).toBeInTheDocument();
      expect(screen.getByText('test')).toBeInTheDocument();
    });
  });
  it('Renders Payslip for Employee and clicks on the row to generate pdf', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: [
        {
          address: 'somewhere',
          amountAccumulated: 2,
          ei: 1,
          eiAccumulated: 2,
          endDate: '2023-01-02T00:00:00',
          fit: 1,
          fitAccumulated: 2,
          grossAmount: 1,
          hoursAccumulated: 2,
          hoursWorked: 1,
          issueDate: '2023-01-05T00:00:00',
          name: 'test',
          netAmount: 1,
          netAccumulated: 3,
          qcpit: 1,
          qcpitAccumulated: 2,
          qpip: 1,
          qpipAccumulated: 2,
          qpp: 1,
          qppAccumulated: 2,
          startDate: '2023-01-01T00:00:00',
        },
      ],
      status: 200,
      statusText: 'OK',
    });

    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/employeedashboard/payslip']}>
        <App />
      </MemoryRouter>
    );
    await waitFor(() => {
      const row = screen.getByText('test');
      user.click(row);
    });

    await waitFor(() => {
      expect(screen.getByText('Payslips')).toBeInTheDocument();
    });
  });
});
