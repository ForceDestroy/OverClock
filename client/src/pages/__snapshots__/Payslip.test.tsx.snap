// Vitest Snapshot v1

exports[`Payslip > Renders Payslip to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
        >
          <div>
            <div
              class="StyledGrid-sc-1wofa1l-0 bySYOJ"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 dDSiKk"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 ibTKOe"
                >
                  <a
                    aria-label="home"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="home"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Home
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="Calendar"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="Calendar"
                    href="/EmployeeDashboard/Schedule"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Calendar
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="MyJob"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="MyJob"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        My Job
                         
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 bqfCTq"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cdHUSp"
                    >
                       
                       
                    </span>
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Sales Associate 
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <a
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    href="/ProfilePage"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 kobaLM"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                      />
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <button
                    class="StyledButton-sc-323bzc-0 kYtlmw"
                    data-testid="logout"
                    type="button"
                  >
                    <svg
                      aria-label="Logout"
                      class="StyledIcon-sc-ofa7kd-0 cnewlt"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                        fill="none"
                        stroke="#231F20"
                        stroke-width="2"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 eETalo"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iNDmPr"
            >
              <h1>
                  Payslips
              </h1>
              <table
                class="StyledTable-sc-1m3u5g-6 YPHTu StyledDataTable-sc-xrlyjm-0 hmFzTO"
              >
                <thead
                  class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4 khLQeI"
                >
                  <tr
                    class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-sc-xrlyjm-2 hSlrxd"
                  >
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Issued For
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Issue Date
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Start Date
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              End Date
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Hours Worked
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Hours Accumulated
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Salary
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              Current Net Pay
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 cwefyf"
                          >
                            <span
                              class="StyledText-sc-1sadyjn-0 hpCRgD"
                            >
                              YTD Net Pay
                            </span>
                          </div>
                        </button>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-sc-xrlyjm-3 kcYfzh"
                  tabindex="0"
                />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
      >
        <div>
          <div
            class="StyledGrid-sc-1wofa1l-0 bySYOJ"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dDSiKk"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ibTKOe"
              >
                <a
                  aria-label="home"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="home"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Home
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="Calendar"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="Calendar"
                  href="/EmployeeDashboard/Schedule"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Calendar
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="MyJob"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="MyJob"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      My Job
                       
                    </span>
                  </div>
                </a>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 bqfCTq"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <span
                    class="StyledText-sc-1sadyjn-0 cdHUSp"
                  >
                     
                     
                  </span>
                  <span
                    class="StyledText-sc-1sadyjn-0 hpCRgD"
                  >
                     Sales Associate 
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <a
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  href="/ProfilePage"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 kobaLM"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                    />
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <button
                  class="StyledButton-sc-323bzc-0 kYtlmw"
                  data-testid="logout"
                  type="button"
                >
                  <svg
                    aria-label="Logout"
                    class="StyledIcon-sc-ofa7kd-0 cnewlt"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                      fill="none"
                      stroke="#231F20"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 eETalo"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 iNDmPr"
          >
            <h1>
                Payslips
            </h1>
            <table
              class="StyledTable-sc-1m3u5g-6 YPHTu StyledDataTable-sc-xrlyjm-0 hmFzTO"
            >
              <thead
                class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4 khLQeI"
              >
                <tr
                  class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-sc-xrlyjm-2 hSlrxd"
                >
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Issued For
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Issue Date
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Start Date
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            End Date
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Hours Worked
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Hours Accumulated
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Salary
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Current Net Pay
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                  <th
                    class="StyledTable__StyledTableCell-sc-1m3u5g-0 ILkGj StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                    scope="col"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 fqUnkf"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 iIPNKk Header__StyledHeaderCellButton-sc-1baku5q-0 jdyvfX"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 cwefyf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            YTD Net Pay
                          </span>
                        </div>
                      </button>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-sc-xrlyjm-3 kcYfzh"
                tabindex="0"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
